flopy/discretization/modeltime.py

```diff
@@ -71,21 +236,367 @@ def totim(self):
+        """
+        Method to create a datetime.datetime object from a variety of user
+        inputs including the following:
+
+        datetime.datetime objects
+        numpy.datetime64 objects
+        pandas.Timestamp objects
+        string objects
+
+        Supported formats for string objects representing November 12th, 2024
+        are as follows:
+
+        '11/12/2024'
+        '11-12-2024'
+        '2024/11/12'
+        '2024-11-12'
+
+        Time can also be represented in the string object. Example formats
+        representing 2:31 pm on November 12th, 2024 are as follows:
+
+        '2024-11-12T14:31:00'
+        '2024/11/12T14:31:00'
+        '11-12-2024t14:31:00'
+        '11/12/2024t14:31:00'
+        '2024-11-12 14:31:00'
+        '2024/11/12 14:31:00'
+        '11-12-2024 14:31:00'
+        '11/12/2024 14:31:00'
+
+        Parameters
+        ----------
+        datetime_obj : various formats
+            a user-supplied representation of date or datetime
+
+        Returns
+        -------
+            datetime.datetime object
+        """
+        if datetime_obj is None:
+            datetime_obj = datetime.datetime(1970, 1, 1)  # unix time zero
+        elif isinstance(datetime_obj, np.datetime64):
+            unix_time_0 = datetime.datetime(1970, 1, 1)
+            ts = (datetime_obj - np.datetime64(unix_time_0)) / np.timedelta64(1, "s")
+            datetime_obj = datetime.datetime.utcfromtimestamp(ts)
+        elif isinstance(datetime_obj, pd.Timestamp):
+            datetime_obj = datetime_obj.to_pydatetime()
+        elif isinstance(datetime_obj, datetime.datetime):
+            pass
+        elif isinstance(datetime_obj, str):
+            str_rep = ModelTime._get_datetime_string_format(datetime_obj)
+            datetime_obj = datetime.datetime.strptime(datetime_obj, str_rep)
+
+        else:
+            raise NotImplementedError(
+                f"{type(datetime_obj)} date representations "
+                f"are not currently supported"
+            )
+
+        return datetime_obj
+
+    def get_totim(self, kper, kstp=None):
+        """
+        Method to get the total simulation time at the end of a given
+        stress period or stress period and time step combination
+
+        Parameters
+        ----------
+        kper : int
+            zero based stress period number
+        kstp : int or None
+            optional zero based time-step number
+
+        Returns
+        -------
+            totim : float
+        """
+        if kstp is None:
+            kstp = self.nstp[kper] - 1
+
+        if not self._totim_dict:
+            self._set_totim_dict()
+
+        if (kper, kstp) not in self._totim_dict:
+            raise KeyError(
+                f"(kper, kstp): ({kper} {kstp}) not a valid combination of "
+                f"stress period and time step"
+            )
+
+        return self._totim_dict[(kper, kstp)]
+
+    def get_datetime(self, kper, kstp=None):
```

Comment:
It might be useful to add an option to return the start datetime of a period/timestep instead. For example, if we are trying to aggregate higher-frequency or mis-aligned data to the model time discretization by computing averages over the model stress periods. Having easy access to the period start and end dates might also be useful for some visualization/post-processing.